sched/init/nx_start.c

```diff
@@ -698,10 +698,12 @@ void nx_start(void)
   /* When control is return to this point, the system is idle. */
 
   sinfo("CPU0: Beginning Idle Loop\n");
+#ifndef CONFIG_DISABLE_IDLE_LOOP
   for (; ; )
     {
       /* Perform any processor-specific idle state operations */
 
       up_idle();
     }
+#endif
```

Comment:
@anjiahao1 why is it necessary to disable the Idle loop?

Reply:
this is because JLINK expect SEGGER_FL_Prepare return and halt CPU to download the image to SRAM, here is the basic workflow:

1. JLINK download nuttx image(contained ofloader) to SRAM through JTAG/SWD
2. JLINK control CPU to run SEGGER_FL_Prepare through JTAG/SWD
3. SEGGER_FL_Prepare call __start and then nx_start->nx_bringup
https://github.com/apache/nuttx-apps/pull/1872/files#diff-54269319ce436252f01744417956400103da2e849e0c59472fde9239886fee6cR187
4. nx_bringup create nsh_main task
5. nsh create ofloader task from /etc/init/rcS
https://github.com/apache/nuttx-apps/pull/1872/files#diff-662f0aa2ce979fda9fa2f94b8e5000208c1f2c6dc8047c8cd81a8f4f00da909cR309
6. When the initialization finish, idle thread call up_idle in an infinite loop

Without this change, SEGGER_FL_Prepare never return, and then JLINK can't start to download the image we want to flash.
With this change, SEGGER_FL_Prepare return, JLINK halt the CPU and download the image through JTAG/SWD, and jump to SEGGER_FL_Program:
https://github.com/apache/nuttx-apps/pull/1872/files#diff-54269319ce436252f01744417956400103da2e849e0c59472fde9239886fee6cR200
SEGGER_FL_Program wake up ofloader task to do the real program.

Reply:
Thank you!